packages/sdk-artifact/src/ArtifactClient.ts

```diff
@@ -24,6 +26,21 @@ export class ArtifactClient {
     return await this.artifactsApi.uploadArtifact(fileStream, size, key, contentType)
   }
 
+  async uploadDirectory(key: string, directoryPath: string): Promise<void> {
+    const allFiles = glob.sync(`${directoryPath}/**/*`, {
+      absolute: true,
+      follow: false,
+      nodir: true,
+    })
+
+    // @todo this could be executed in parallel
```

Comment:
This could fail in the middle and leave unconsistent state. Def not in the scope of this PR, but something to have in mind